### PR TITLE
In-house Shipping

### PR DESCRIPTION
**Pull request by rithiksharma100**

### CODE REVIEW

File: includes/endpoints/internal/class-simpl-checkout-order-controller.php [hunk 1]
```diff
@@ -63,18 +59,16 @@ function create(WP_REST_Request $request)
         }
     }
 
-    protected function update_order_metadata($request, $order)
```
Comment (piyush-getsimpl) [inline includes/endpoints/internal/class-simpl-checkout-order-controller.php:66]: Lets use simpl_ prefix for all the functions/class we create going forward

File: includes/endpoints/internal/class-simpl-checkout-order-controller.php [hunk 2]
```diff
@@ -63,18 +59,16 @@ function create(WP_REST_Request $request)
...
-        $order->update_meta_data("simpl_cart_token", $request->get_params()["simpl_cart_token"]);
-        $order->update_meta_data("simpl_payment_id", $request->get_params()["simpl_payment_id"]);
-        $order->update_meta_data("simpl_order_id", $request->get_params()["simpl_order_id"]);
-        if ($request->get_params()["simpl_payment_type"] == 'Cash on Delivery (COD)') {
-            $order->set_payment_method('cod');
-            $order->set_payment_method_title('Cash on delivery');
-        } else {
-            $order->set_payment_method('Simpl Checkout');
-            $order->set_payment_method_title($request->get_params()["simpl_payment_type"]);
+    static function update_order_details($request, $order) {
```
Comment (piyush-getsimpl) [inline includes/endpoints/internal/class-simpl-checkout-order-controller.php:62]: Lets use simpl_ prefix for all the functions/class we create going forward

File: includes/helpers/class-simpl-wc-helper.php [hunk 3]
```diff
@@ -86,7 +86,30 @@ static function set_address_in_cart($shipping_address, $billing_address) {
         }
     }
 
-    static function set_utm_info_in_order($request, $order) {
+    static function update_order_metadata($request, $order)
```
Comment (piyush-getsimpl) [inline includes/helpers/class-simpl-wc-helper.php:89]: simpl_update_order_metadata

File: includes/helpers/class-simpl-wc-helper.php [hunk 4]
```diff
@@ -86,7 +86,30 @@ static function set_address_in_cart($shipping_address, $billing_address) {
...
+            self::set_utm_info_in_order($request, $order);
+        }
+    }
+
+    static protected function is_utm_info_present($request)
+    {
+        return (isset($request->get_params()["utm_info"]) && count($request->get_params()["utm_info"]) > 0);
+    }
+
+    static protected function set_utm_info_in_order($request, $order) {
```
Comment (piyush-getsimpl) [inline includes/helpers/class-simpl-wc-helper.php:112]: simpl_set_utm_info_in_order

File: includes/helpers/class-simpl-wc-helper.php [hunk 5]
```diff
@@ -190,6 +212,18 @@ static protected function create_new_customer($order) {
 
         return $customer;
     }
+
+    static function set_shipping_method_in_order($order, $shipping_method) {
```
Comment (piyush-getsimpl) [inline includes/helpers/class-simpl-wc-helper.php:216]: simpl_set_shipping_method_in_order

File: includes/endpoints/internal/class-simpl-checkout-order-controller.php (hunk 2 above)
Comment (rithiksharma100) [reply includes/endpoints/internal/class-simpl-checkout-order-controller.php]: done

File: includes/helpers/class-simpl-wc-helper.php (hunk 3 above)
Comment (rithiksharma100) [reply includes/helpers/class-simpl-wc-helper.php]: done

File: includes/helpers/class-simpl-wc-helper.php (hunk 4 above)
Comment (rithiksharma100) [reply includes/helpers/class-simpl-wc-helper.php]: done

File: includes/helpers/class-simpl-wc-helper.php (hunk 5 above)
Comment (rithiksharma100) [reply includes/helpers/class-simpl-wc-helper.php]: done

File: simpl-checkout-woocommerce.php [hunk 6]
```diff
@@ -107,6 +107,7 @@ function simpl_checkout_int() {
     include_once 'includes/plugin_support/load.php';
 
     add_filter( 'woocommerce_payment_gateways', 'simpl_add_gateway_class' );
+    add_filter( 'woocommerce_shipping_chosen_method', '__return_false', 99); // this disables the application of default shipping method
```
Comment (sathia27) [inline simpl-checkout-woocommerce.php:110]: Could this potentially create problems for the merchant? For instance, if our plugin, once installed, disables WooCommerce's shipping functionality, what happens when a customer want to use native woocommerce?

**Review by sathia27 — CHANGES_REQUESTED** [review]

File: simpl-checkout-woocommerce.php (hunk 6 above)
Comment (rithiksharma100) [reply simpl-checkout-woocommerce.php:110]: @sathia27 I tested this out on our staging store, the only change that happens is that the default shipping method is not selected in the native flow, so customers will have to select one manually

Comment (rithiksharma100) [reply simpl-checkout-woocommerce.php:110]: i'll do prod sanity as well on our prod store before installing this with merchants

File: includes/helpers/class-simpl-wc-helper.php [hunk 7]
```diff
@@ -86,7 +86,30 @@ static function set_address_in_cart($shipping_address, $billing_address) {
         }
     }
 
-    static function set_utm_info_in_order($request, $order) {
+    static function simpl_update_order_metadata($request, $order)
```
Comment (sathia27) [inline includes/helpers/class-simpl-wc-helper.php:89]: we are planning to go public, let's start writing tests for plugin.

**Review by sathia27 — CHANGES_REQUESTED** [review]

File: includes/helpers/class-simpl-wc-helper.php (hunk 7 above)
Comment (rithiksharma100) [reply includes/helpers/class-simpl-wc-helper.php:89]: sure

Comment (rithiksharma100) [reply includes/helpers/class-simpl-wc-helper.php:89]: discussed with @piyush-getsimpl, it's taking a lot of time to write test cases, so I'll release the test cases in a subsequent PR

File: includes/helpers/class-simpl-wc-helper.php [hunk 8]
```diff
@@ -86,7 +86,30 @@ static function set_address_in_cart($shipping_address, $billing_address) {
...
+    {
+        $order->update_meta_data("simpl_cart_token", $request->get_params()["simpl_cart_token"]);
+        $order->update_meta_data("simpl_payment_id", $request->get_params()["simpl_payment_id"]);
+        $order->update_meta_data("simpl_order_id", $request->get_params()["simpl_order_id"]);
+        if ($request->get_params()["simpl_payment_type"] == 'Cash on Delivery (COD)') {
```
Comment (sathia27) [inline includes/helpers/class-simpl-wc-helper.php:94]: Can you move these to constant `Cash on Delivery (COD)`

File: includes/helpers/class-simpl-wc-helper.php [hunk 9]
```diff
@@ -86,7 +86,30 @@ static function set_address_in_cart($shipping_address, $billing_address) {
...
+            $order->set_payment_method('cod');
+            $order->set_payment_method_title('Cash on delivery');
+        } else {
+            $order->set_payment_method('Simpl Checkout');
```
Comment (sathia27) [inline includes/helpers/class-simpl-wc-helper.php:98]: this also can move to constant. This is right time to fix these existing issues.

**Review by sathia27 — CHANGES_REQUESTED** [review]

File: includes/helpers/class-simpl-wc-helper.php (hunk 9 above)
Comment (rithiksharma100) [reply includes/helpers/class-simpl-wc-helper.php]: done

**Review by sathia27 — DISMISSED** [review]

**Review by sathia27 — APPROVED** [review]

**Pull request merged**